include/drivers/gpio.h

```diff
@@ -226,13 +988,49 @@ static inline int gpio_pin_write(struct device *port, u32_t pin,
  * @param pin Pin number where data is read.
  * @param value Integer pointer to receive the data values from the pin.
  * @return 0 if successful, negative errno code on failure.
+ *
+ * @deprecated This function has been deprecated, please use gpio_pin_get(),
+ * gpio_pin_get_raw() instead.
  */
-static inline int gpio_pin_read(struct device *port, u32_t pin,
+__deprecated static inline int gpio_pin_read(struct device *port, u32_t pin,
 				u32_t *value)
 {
 	return gpio_read(port, GPIO_ACCESS_BY_PIN, pin, value);
 }
 
+/**
+ * @brief Configure pin interrupt.
+ *
+ * @note This function can also be used to configure interrupts on pins
+ *       not controlled directly by the GPIO module. That is, pins which are
+ *       routed to other modules such as I2C, SPI, UART.
+ *
+ * @param port Pointer to device structure for the driver instance.
+ * @param pin Pin number.
+ * @param flags Interrupt configuration flags as defined by GPIO_INT_*.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP if any of the configuration options is not supported.
+ * @retval -EINVAL Invalid argument.
+ */
```

Comment:
Some hardware has limited number of interrupt lines that are shared between different GPIO ports. I will add `-EACCES` "Permission denied" error code to indicate that it's not possible to configure interrupt for a given pin if the hardware resource is already taken.
```
 * @retval -EACCES Interrupt line required to configure pin interrupt is already in use.
```
Another error code that we possibly could use is -EPERM "Operation not permitted".

Reply:
`-EBUSY` "Device or resource busy" would also be appropriate.

Reply:
Yes, that's probably a better choice. I'll use `-EBUSY`.